py/torch_tensorrt/dynamo/aten_tracer.py

```diff
@@ -0,0 +1,157 @@
+        specialize_int: bool = True,
+        verbose: bool = True,
+    ) -> None:
+
+        self.capture_scalar_outputs = capture_scalar_outputs
+        self.guard_nn_modules = guard_nn_modules
+        self.dynamic_shapes = dynamic_shapes
+        self.specialize_int = specialize_int
+        self.verbose = verbose
+
+    def activate(self) -> None:
+        torchdynamo.config.capture_scalar_outputs = self.capture_scalar_outputs
+        torchdynamo.config.guard_nn_modules = self.guard_nn_modules
+        torchdynamo.config.dynamic_shapes = self.dynamic_shapes
+        torchdynamo.config.specialize_int = self.specialize_int
+        torchdynamo.config.verbose = self.verbose
+
+    def deactivate(self) -> None:
+        torchdynamo.config.capture_scalar_outputs = True
+        torchdynamo.config.guard_nn_modules = True
+        torchdynamo.config.dynamic_shapes = True
+        torchdynamo.config.specialize_int = True
+        torchdynamo.config.verbose = True
+
+
+@contextmanager
+def using_config(config: DynamoConfig) -> Generator[DynamoConfig, None, None]:
+    config.activate()
+    try:
+        yield config
+    finally:
+        config.deactivate()
+
+
+@contextmanager
+def setting_python_recursive_limit(limit: int = 10000) -> Generator[None, None, None]:
+    """
+    Temporarily increase the python interpreter stack recursion limit.
+    This is mostly used for pickling large scale modules.
+    """
+    default = sys.getrecursionlimit()
+    if limit > default:
+        sys.setrecursionlimit(limit)
+    try:
+        yield
+    finally:
+        sys.setrecursionlimit(default)
+
+
+@req_torch_version("2.dev")
+def dynamo_trace(
+    f: Callable[..., Value],
+    # pyre-ignore
+    args: Tuple[Any, ...],
+    aten_graph: bool,
+    tracing_mode: str = "real",
+    dynamo_config: Optional[DynamoConfig] = None,
+) -> Tuple[torch.fx.GraphModule, Set]:
+    """
+    TODO: Once we fully migrate to torchdynamo frontend, we will remove
+    this config option alltogether.  For now, it helps with quick
+    experiments with playing around with TorchDynamo
+    """
+    if dynamo_config is None:
+        dynamo_config = DynamoConfig()
+    with using_config(dynamo_config), setting_python_recursive_limit(2000):
+        torchdynamo.reset()
+        try:
+            return torchdynamo.export(
+                f,
+                *copy.deepcopy(args),
+                aten_graph=aten_graph,
+                tracing_mode=tracing_mode,
+            )
+        except torchdynamo.exc.Unsupported as exc:
+            raise RuntimeError(
+                "The user code is using a feature we don't support. "
+                "Please try torchdynamo.explain() to get possible the reasons",
+            ) from exc
+        except Exception as exc:
+            raise RuntimeError(
+                "torchdynamo internal error occured. Please see above stacktrace"
+            ) from exc
+
+
+@req_torch_version("2.dev")
+def trace(model, inputs, **kwargs):
```

Comment:
Thoughts on exposing this to users  under `torch_tensorrt.dynamo.trace`?

Reply:
I added it now. The final version is likely going to be same as calling `torch._dynamo.export` in which case we might not need to expose it.